scripts/experiments_tabular/vae.jl

```diff
@@ -0,0 +1,142 @@
+using DrWatson
+@quickactivate
+using ArgParse
+using GenerativeAD
+import StatsBase: fit!, predict
+using StatsBase
+using BSON
+using Flux
+
+s = ArgParseSettings()
+@add_arg_table! s begin
+   "max_seed"
+        required = true
+        arg_type = Int
+        help = "seed"
+    "dataset"
+        required = true
+        arg_type = String
+        help = "dataset"
+end
+# for testing purposes
+if length(ARGS) == 0
+	push!(ARGS, "1")
+	push!(ARGS, "iris")
+end
+parsed_args = parse_args(ARGS, s)
+@unpack dataset, max_seed = parsed_args
+
+#######################################################################################
+################ THIS PART IS TO BE PROVIDED FOR EACH MODEL SEPARATELY ################
+modelname = "vae"
+# sample parameters, should return a Dict of model kwargs 
+"""
+	sample_params()
+
+Should return a named tuple that contains a sample of model parameters.
+"""
+function sample_params()
+	par_vec = (2 .^(3:8), 2 .^(4:9), 10f0 .^(-4:-3), 2 .^ (5:7), ["relu", "swish", "tanh"], 3:4, 1:Int(1e8))
+	argnames = (:zdim, :hdim, :lr, :batchsize, :activation, :nlayers, :init_seed)
+	parameters = (;zip(argnames, map(x->sample(x, 1)[1], par_vec))...)
+	# ensure that zdim < hdim
+	while parameters.zdim >= parameters.hdim
```

Comment:
What about putting this into `edit_params`, where you can divide by 2 until the condition is satisfied?

Reply:
I don't like dividing it by two since then it is truly random anymore - higher values closer to hdim will tend to get selected more often. Also, I would have to redefine the sampling range for the parameter again in `edit_params` which I want to avoid since it would be easy to edit it in once place and not in the other.

Reply:
Makes sense.